### PR TITLE
Thermopro TP-11.

### PR DESCRIPTION
**Pull request by pekon**
I tested this with my device. I'll send a separate pull request with the test files.

### CODE REVIEW

**Comment by zuckschwerdt** [comment]
Add new devices at the end, otherwise existing IDs are invalidated (`-R` flag).
Also `bitbuffer_find_repeated_row()` might be useful.

**Comment by pekon** [comment]
Thanks! bitbuffer_find_repeated_row assumes that all rows have exactly the same number of bits. For this device the first three rows have 33 bits and the last has 32 bits. This code only compares the first 32 bits.

**Comment by zuckschwerdt** [comment]
good point, that needs to be fixed someday. `compare_rows()` should return a prefix length which can be matched up with `min_bits`.

**Comment by zuckschwerdt** [comment]
One more thing: move the `local_time_str()` call down to just before `data_make()`.

**Comment by pekon** [comment]
Done. Also made sure the code exits as soon as it detects that no more than two rows can be same.

File: src/devices/thermopro_tp11.c [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+                && bb[i][1] == bb[j][1]
+                && bb[i][2] == bb[j][2]
+                && bb[i][3] == bb[j][3]
+                && ++equal_rows >= 2) good = i;
+        }
+    }
+    if (good < 0) return 0;
+    // bb[good] is equal to at least two other rows, decode.
+    unsigned value = (bb[good][0] << 16) + (bb[good][1] << 8) + bb[good][2];
+    unsigned device = value >> 12;
```
Comment (merbanan) [inline src/devices/thermopro_tp11.c]: Move the declaration to the top, and I prefer unsigned int over just unsigned.

**Comment by merbanan** [comment]
Looks good. One tiny change and samples before I merge.

**Comment by pekon** [comment]
Done. Thanks!

**Pull request merged**